npm/pkg/transport/pb/transport.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+package pb;
+option go_package = "github.com/Azure/azure-container-networking/npm/pkg/transport/pb;pb";
+
+import "google/protobuf/struct.proto";
+
+// DataplaneEvents represents the Service RPC exposed by the gRPC server.
+service DataplaneEvents{
+	rpc Connect(DatapathPodMetadata) returns (stream Events);
+}
+
+// DatapathPodMetadata is the metadata for a datapath pod
+message DatapathPodMetadata {
+  string pod = 1; // Daemonset Pod ID
+  string node = 2; // Node name
+}
+
+// Events defines the operation (event type) and object type being
+// streamed to the datapath client. A events message may carry one or
+// more Event objects.
+message Events{
+  EventType type = 1;
+  // The following events are supported:
+  enum EventType {
+    // Send a CREATE/UPDATE event to a datapath Pod
+    APPLY = 0;
+    // Send a DELETE event to a datapath Pod
+    REMOVE = 1;
+  }
+
+  ObjectType object = 2;
+  // Supported objects are IPSet rules and NetworkPolicy objects.
+  enum ObjectType {
+    IPSET = 0;
+    POLICY = 1;
+  }
+  // Payload can contain one or more Event objects.
+  repeated Event event = 3;
+}
+
+// Event is a generic object that can be Created, 
+// Updated, Deleted by the controlplane.
+message Event {
+  // Data can container one or more instances of IPSet or NetworkPolicy
+  // objects.
+	repeated google.protobuf.Struct data = 3;
```

Comment:
I'm guessing this is still in prototyping stages, but I've usually found working with `Struct` difficult. Is the structure of IPSet and NetworkPolicy well defined? Could use `oneof`

Reply:
I am afraid that we might get out of sync if things change where someone might forget to update the proto file

Reply:
make updating the proto part of the build and fail if it diffs dirty

Reply:
How do you diff it with a standard go struct? The proto generated structure will most likely vary from the existing objects due to lack of type support.

Reply:
you would diff what the build generates vs what's checked in. this ensures that go proto code is up to date with the source of truth, the proto file.

Reply:
I still don't think I understand how `Struct` helps in this case. For prototyping purposes it should be fine, but it's a lot easier to for client and server to get out of sync when using it since it is basically like json and foregoes the benefits of strong typing.

Reply:
> diff what the build generates vs what's checked in

to expand, you `make gen` and then something like `test -z "$(git status --porcelain)"` which returns 0 if the pwd is clean and 1 if there are changes vs source

Reply:
> you would diff what the build generates vs what's checked in. this ensures that go proto code is up to date with the source of truth, the proto file.

I don't mean the proto generated code. I mean the actual IPSet, and other structs defined in `types.go`.

Reply:
> I still don't think I understand how `Struct` helps in this case. For prototyping purposes it should be fine, but it's a lot easier to for client and server to get out of sync when using it since it is basically like json and foregoes the benefits of strong typing.

I agree. This is not set in stone. Good for testing. I will play around with it and figure out what works best.

Reply:
@nitishm  once we build a goatStateProcessor with the type generated from this proto file then if proto generation gets dirty, GSProcessor build should fail (ideally). Lets move forward at this moment with current "struct" design, this is a 2-way door, as we build bits around this, i am sure we will tweak this struct further, so it should be okay with current type. Wdygt ?